### PR TITLE
x86: disable GRA for vectors & masks

### PR DESCRIPTION
**Pull request by BradleyWood**
Signed-off-by: BradleyWood <bradley.wood@ibm.com>

### CODE REVIEW

**Comment by BradleyWood** [comment]
jenkins build all

File: compiler/x/codegen/OMRTreeEvaluator.hpp [hunk 1]
```diff
@@ -396,6 +396,10 @@ class OMR_EXTENSIBLE TreeEvaluator: public OMR::TreeEvaluator
                                            TR::CodeGenerator *cg,
                                            TR::Instruction *currentInstruction = NULL);
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRTreeEvaluator.hpp:398]: Can you add Doxygen comments for these new APIs please?

Comment (BradleyWood) [reply compiler/x/codegen/OMRTreeEvaluator.hpp]: These aren't new APIs, which made me realize that they belong to OMR::CodeGenerator. Anyway, I added docs to the parent definition.

File: compiler/x/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -3227,3 +3227,28 @@ OMR::X86::CodeGenerator::directCallRequiresTrampoline(intptr_t targetAddress, in
...
+   {
+   return !(dt.isVector() || dt.isMask());
+   }
+
+bool
+OMR::X86::CodeGenerator::considerTypeForGRA(TR::SymbolReference *symRef)
+   {
+   if (symRef && symRef->getSymbol())
+      {
+      return considerTypeForGRA(symRef->getSymbol()->getDataType());
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRCodeGenerator.cpp:3248]: This should be `self()->considerTypeForGRA( ... );`

**Comment by BradleyWood** [comment]
jenkins build all

**Comment by 0xdaryl** [comment]
Jenkins build win

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**